pkg/build/webhook/generic/generic.go

```diff
@@ -54,13 +52,19 @@ func (p *WebHookPlugin) Extract(buildCfg *api.BuildConfig, secret, path string,
 	}
 
 	if req.Body != nil && contentType == "application/json" {
+		// REVIEW: a real world scenario is that we don't use the same request to
+		// verify a secret. if that would be the case, then the req.body is already
+		// consumed for that request after the first try and we return proceed=true
+		// here even when the git ref doesn't match (which shouldn't happen).
```

Comment:
i was ambiguous in this note. i thought we might reuse a request somewhere else, but i don't see a real case scenario. you have a webhook request, you trigger a build or fail if it doesn't match the ref/secret. it's that simple. i was over thinking it